### PR TITLE
EIP-1559 RPC changes

### PR DESCRIPTION
**Pull request by lightclient**
This branch contains EIP-1559 specific changes to the RPC, specifically:
* eth_gasPrice
* dynamicFeeTx can be displayed in RPC response
* effectiveGasPrice is computed on-the-fly for eth_getTransaction*
* add baseFee to RPC headers

A few things that haven't been addressed yet:
* eth_estimateGas does not account for dynamicFeeTxs effective gas price vs. EOA balance
* gas price oracle does not gracefully handle 1559 fork, it naively finds the median gas price of the previous `n` blocks

### CODE REVIEW

**Comment by holiman** [comment]
Trying to submit a transaction to `baikal` on this PR, I get
```
WARN [05-17|13:20:37.421] Served eth_sendTransaction               reqid=22 t=2.90379ms  err="tip higher than fee cap"
```

**Comment by holiman** [comment]
Hm, also, 
```
> runcode("0x61efef6000526010601ff3")
Error: invalid code: must not begin with 0xef
	at web3.js:6347:37(47)
	at web3.js:5081:62(37)
	at baikal.js:5:23(20)
	at <eval>:1:8(2)
> 
```
```
runcode = function(code){
   tx = {from: eth.accounts[0], input: code, tip:"0x1", feeCap: "0x10"}
   eth.sendTransaction(tx)
}
```
It's quirky that it won't allow me to create the tx in that case

**Comment by lightclient** [comment]
It appears the issue is that because there are no txs for the gas price oracle to use to calculate a good tip, it uses the client default (`1000000000`) which is much higher than the recommended `feeCap` of `0x0a` (2x the last block `baseFee). I'll update so that `feeCap = min(2*parent.BaseFee(), tip)`.

**Comment by lightclient** [comment]
> It's quirky that it won't allow me to create the tx in that case

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -1156,6 +1178,12 @@ func RPCMarshalHeader(head *types.Header) map[string]interface{} {
 		"transactionsRoot": head.TxHash,
 		"receiptsRoot":     head.ReceiptHash,
 	}
+
+	if head.BaseFee != nil {
+		result["baseFee"] = (*hexutil.Big)(head.BaseFee)
```
Comment (ryanschneider) [inline internal/ethapi/api.go:1183]:
```suggestion
		result["baseFeePerGas"] = (*hexutil.Big)(head.BaseFee)
```

**Pull request closed without merging**